pepys_timeline/db.py

```diff
@@ -23,30 +26,38 @@ def get_db_conn_kwargs():
     return db_params
 
 
-def get_config_options():
+def get_query_result(query, vars_=None):
     db_conn_kwargs = get_db_conn_kwargs()
-    with psycopg2.connect(**db_conn_kwargs) as conn:
-        with conn.cursor(cursor_factory=RealDictCursor) as curs:
-            curs.execute(CONFIG_OPTIONS_QUERY)
-            config_options = curs.fetchall()
-    return config_options
+    try:
+        with psycopg2.connect(**db_conn_kwargs) as conn:
+            try:
+                with conn.cursor(cursor_factory=RealDictCursor) as curs:
+                    curs.execute(query, vars_)
+                    result = curs.fetchall()
+            except psycopg2.Error:
+                raise DatabaseQueryError('Error querying database.')
```

Comment:
@krsnik93 is the actual error logged anywhere?
If it isn't easily accessible then I suggest it gets included in the message, so the users can pass it on to us.

Reply:
@IanMayo Very good point! Great moment to add logging to the app. I have set it so that it logs to stdout, which means it will be easily accesible in Heroku logs. If we ever wish to log to a file, the change will be simple.

Reply:
Thanks @krsnik93 

Yes - I recognise the value of Heroku logs.

But, my users may not know how to access stdout logging.  If any of the connection/database errors happen the impact is catastrophic - it just won't work.  So, in my opinion it's fine to show the errors there in the browser-page. Then the users can most quickly/effectively pass the error onto me (or their tech support), and we can get it fixed.  

So, can we show the errors in the browser page please?

Reply:
@IanMayo Gotcha, sorry for the earlier confusion. I have made a change to show the whole exception traceback in the error field in the GUI.

Reply:
![image](https://user-images.githubusercontent.com/13034472/118531507-d0dd8a80-b73d-11eb-9308-bf8a8afbec42.png)

Reply:
Thanks. Minor improvement, please @krsnik93 

Could we have `Error connecting to the database` as `h3` (or something like that) then the traceback as `normal` (or something like that) please?

This will inform the users what has happened in general terms, then offer more detail for follow-up.

Reply:
@IanMayo Good point. Let me know if the paragraph does not look visible enough.

![image](https://user-images.githubusercontent.com/13034472/118535146-02f0eb80-b742-11eb-9422-dca35ca46f42.png)

Reply:
Could you translate the newlines `\n` in the Python string into newlines (`<br>` or something) in the HTML, so that the traceback displays properly? Thanks!

Reply:
Aah, this _may_ introduce the need for scrollbars in the new `div`.

Reply:
Good point, sorry about that.

![image](https://user-images.githubusercontent.com/13034472/118537483-e86c4180-b744-11eb-9efa-3ec0611062c8.png)

@IanMayo It does overflow on my screen and shows a scrollbar, would this be an issue? We could try reducing font or clipping or maybe shortening the message...

Reply:
Thanks @krsnik93  the scroll-bar is acceptable.